### PR TITLE
07. Test "Categories" dropdown list

### LINKED ISSUE #12: 07. Test "Categories" dropdown list

### PR DESCRIPTION
**Pull request by maksym-bielyshev**

### CODE REVIEW

File: tests_data/category_dropdown_list.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ua;'Будь-який', 'Легкові', 'Мото', 'Вантажівки', 'Причепи', 'Спецтехніка', 'Сільгосптехніка', 'Автобуси', 'Водний транспорт', 'Повітряний транспорт', 'Автобудинки'
```
Comment (denyskulykov) [inline tests_data/category_dropdown_list.csv]: Please reuse files tests_data/category_dropdown.csv and second one

File: tests/test_home_page.py [hunk 2]
```diff
@@ -56,6 +54,28 @@ def test_categories_dropdown_transition_without_naming(
...
+
+        Steps:
+            1. Change language to a proper language
+            2. Click on the "Categories" dropdown
+            3. Check if an option in the expected list.
+
+        Expected result: all options are in the expected list.
+        """
+        BasePage(self.driver).switch_proper_language(language)
+        categories_dropdown = self.driver.find_element(*LocatorsHomeFilter.
```
Comment (denyskulykov) [inline tests/test_home_page.py]: Move to page object

File: tests/test_home_page.py [hunk 3]
```diff
@@ -56,6 +54,28 @@ def test_categories_dropdown_transition_without_naming(
         """
         BasePage(self.driver).switch_proper_language(language)
         self.home_page.category_dropdown.choose_dropdown_option(category)
-        self.home_page = HomePage(self.driver)
         self.home_page.click_search_button()
         assert substring in self.driver.current_url
+
+    @pytest.mark.parametrize(
```
Comment (denyskulykov) [inline tests/test_home_page.py:54]: Add test for duplication like this:
'Легкові'
'Легкові'

File: tests/test_home_page.py [hunk 4]
```diff
@@ -56,6 +53,41 @@ def test_categories_dropdown_transition_without_naming(
...
+
+    @pytest.mark.parametrize('language, expected_categories, _',
+                             get_test_data_dictreader(
+                                 'category_dropdown.csv',
+                                 'category_dropdown_without_naming.csv'))
+    def test_no_doubles_dropdown_categories(self, language,
+                                            expected_categories, _):
+        actual_categories = []
+        self.home_page.switch_proper_language(language)
+        for category in self.home_page.get_all_categories():
```
Comment (denyskulykov) [inline tests/test_home_page.py]: actual_categories = [c.text for c in self.home_page.get_all_categories()]

File: tests/test_home_page.py [hunk 5]
```diff
@@ -56,6 +53,41 @@ def test_categories_dropdown_transition_without_naming(
...
+        Steps:
+            1. Change language to a proper language
+            2. Click on the "Categories" dropdown
+            3. Check if an option in the expected list.
+
+        Expected result: all options are in the expected list.
+        """
+        self.home_page.switch_proper_language(language)
+        actual_categories = []
+        for category in self.home_page.get_all_categories():
```
Comment (denyskulykov) [inline tests/test_home_page.py]: the same

File: tests/test_home_page.py (hunk 4 above)
Comment (denyskulykov) [reply tests/test_home_page.py]: len(set(list)) == len(list)

File: tests/test_home_page.py [hunk 6]
```diff
@@ -56,6 +53,41 @@ def test_categories_dropdown_transition_without_naming(
...
+            actual_categories.append(category.text)
+        assert expected_categories in actual_categories
+
+    @pytest.mark.parametrize('language, expected_categories, _',
+                             get_test_data_dictreader(
```
Comment (denyskulykov) [inline tests/test_home_page.py]: remove get data from files

File: tests/test_home_page.py [hunk 7]
```diff
@@ -56,6 +53,41 @@ def test_categories_dropdown_transition_without_naming(
...
         assert substring in self.driver.current_url
+
+    @pytest.mark.parametrize(
+        'language, expected_categories, _',
+        get_test_data_dictreader('category_dropdown.csv',
+                                 'category_dropdown_without_naming.csv'))
+    def test_categories_dropdown_options_list(self, language,
+                                              expected_categories, _) -> None:
+        """Check if "Categories" dropdown list is equal to expected list
+        in Ukrainian.
```
Comment (denyskulykov) [inline tests/test_home_page.py]: remove Ukrainian

File: tests/test_home_page.py (hunk 6 above)
Comment (maksym-bielyshev) [reply tests/test_home_page.py:82]: Changed to get only language

Comment (maksym-bielyshev) [reply tests/test_home_page.py]: Changed to get language

**Pull request merged**